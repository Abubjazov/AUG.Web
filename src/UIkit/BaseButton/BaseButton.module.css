.root {
  display: flex;
  align-items: center;
  justify-content: center;
  height: 40px;
  padding: 6px 20px;
  font-size: 12px;
  font-weight: 500;
  line-height: 12px;
  text-align: center;
  text-transform: uppercase;
  cursor: pointer;
  border: 0;
  border-radius: 10px;
  transition: all var(--transition-delay) linear;
}

.outlined-white {
  color: var(--white);
  background-color: transparent;
  border: 1px solid var(--white);
}

.contained-blue {
  color: var(--white);
  background-color: var(--blue);
}

.contained-red {
  color: var(--white);
  background-color: var(--red);
}

.disabled {
  color: var(--white);
  cursor: not-allowed;
  background-color: var(--gray);
  border: 0;
}

.skeleton {
  width: 60px;
  padding: 0 20px;
  cursor: not-allowed;
  opacity: 0.5;
  animation: skeleton-loading 1s linear infinite alternate;
}

@keyframes skeleton-loading {
  0% {
    background-color: hsl(198deg 33% 85%);
  }

  100% {
    background-color: hsl(200deg 10% 95%);
  }
}

@media (hover: hover) {
  .outlined-white:hover {
    color: var(--textgray);
    background-color: var(--white-o);
    border: 1px solid var(--white-o);
  }

  .contained-blue:hover {
    color: var(--blue);
    background-color: var(--white-o);
  }

  .contained-red:hover {
    color: var(--red);
    background-color: var(--white-o);
  }
}
