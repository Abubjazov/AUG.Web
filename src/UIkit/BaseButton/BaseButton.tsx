import { FC } from 'react'

import Spinner from 'uikit/Spinner/Spinner'
import { combineClasses as cc } from 'utils/combineClasses/combineClasses'

import styles from './BaseButton.module.css'

export enum EBaseButtonMode {
  OUTLINED_WHITE = 'outlined-white',
  CONTAINED_BLUE = 'contained-blue',
  CONTAINED_RED = 'contained-red',
}

export interface BaseButtonProps {
  userStyles?: string
  loading?: boolean
  disabled?: boolean
  mode?: EBaseButtonMode
  label: string
  onClick?: () => void
}

const BaseButton: FC<BaseButtonProps> = ({
  userStyles = '',
  loading = false,
  disabled = false,
  mode = EBaseButtonMode.OUTLINED_WHITE,
  label,
  onClick,
}) => {
  return (
    <button
      type="button"
      data-testid="base-button"
      disabled={disabled}
      className={cc([
        styles.root,
        styles[mode],
        `${disabled ? styles.disabled : ''}`,
        userStyles,
      ])}
      onClick={onClick}
    >
      {loading ? (
        <Spinner width={20} height={20} strokeWidth={8} strokeColor="#fff" />
      ) : (
        label
      )}
    </button>
  )
}

export default BaseButton
