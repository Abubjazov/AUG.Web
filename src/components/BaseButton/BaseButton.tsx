import { FC } from 'react'

import styles from './BaseButton.module.css'

export enum BaseButtonMode {
  OUTLINED = 'outlined',
  CONTAINED_BLUE = 'contained-blue',
  CONTAINED_RED = 'contained-red',
}

export interface BaseButtonProps {
  loading?: boolean
  disabled?: boolean
  mode?: BaseButtonMode
  label: string
  onClick?: () => void
}

const BaseButton: FC<BaseButtonProps> = ({
  loading = false,
  disabled = false,
  mode = BaseButtonMode.OUTLINED,
  label,
  onClick,
}) => {
  return loading ? (
    <div className={[styles.root, styles.skeleton].join(' ')}></div>
  ) : (
    <button
      type="button"
      data-testid="base-button"
      disabled={disabled}
      className={[
        styles.root,
        styles[mode],
        `${disabled ? styles.disabled : ''}`,
      ].join(' ')}
      onClick={onClick}
    >
      {label}
    </button>
  )
}

export default BaseButton
