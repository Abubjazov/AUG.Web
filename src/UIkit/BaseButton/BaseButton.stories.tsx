import type { Meta, StoryObj } from '@storybook/react'

import BaseButton, { BaseButtonProps } from './BaseButton'

const meta: Meta<BaseButtonProps> = {
  component: BaseButton,
  title: 'Components/Base Button',
  parameters: {
    docs: {
      description: {
        component: 'Standard interface component - basic button',
      },
    },
  },
  tags: ['autodocs'],
  argTypes: {
    label: {
      description: 'Button contents',
    },
    mode: {
      description: 'Button appearance',
      defaultValue: 'outlined',
    },
    disabled: {
      description: 'Button is not active',
    },
    loading: {
      description: 'Appearance of the button when loading content (skeleton)',
    },
    onClick: {
      description: 'Optional click handler',
    },
  },
}

export default meta

type Story = StoryObj<BaseButtonProps>

export const Default: Story = {
  args: {
    label: 'Base Button',
  },
}
