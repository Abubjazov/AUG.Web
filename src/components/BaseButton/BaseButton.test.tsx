import { render } from '@testing-library/react'

import BaseButton from '.'

describe('BaseButton', () => {
  test('should render BaseButton default', () => {
    expect(render(<BaseButton label={'Base Button'} />)).toMatchSnapshot()
  })

  test('should render BaseButton skeleton', () => {
    expect(
      render(<BaseButton loading label={'Base Button'} />),
    ).toMatchSnapshot()

    expect(
      render(<BaseButton loading disabled label={'Base Button'} />),
    ).toMatchSnapshot()

    expect(
      render(
        <BaseButton
          loading
          disabled
          mode="contained-blue"
          label={'Base Button'}
        />,
      ),
    ).toMatchSnapshot()

    expect(
      render(<BaseButton loading mode="contained-red" label={'Base Button'} />),
    ).toMatchSnapshot()
  })

  test('should render BaseButton disabled', () => {
    expect(
      render(<BaseButton disabled label={'Base Button'} />),
    ).toMatchSnapshot()

    expect(
      render(<BaseButton disabled mode="outlined" label={'Base Button'} />),
    ).toMatchSnapshot()
  })

  test('should render BaseButton mode: "outlined"', () => {
    expect(
      render(<BaseButton mode="outlined" label={'Base Button'} />),
    ).toMatchSnapshot()
  })

  test('should render BaseButton mode: "outlined"', () => {
    expect(
      render(<BaseButton mode="outlined" label={'Base Button'} />),
    ).toMatchSnapshot()
  })

  test('should render BaseButton mode: "contained-blue"', () => {
    expect(
      render(<BaseButton mode="contained-blue" label={'Base Button'} />),
    ).toMatchSnapshot()
  })

  test('should render BaseButton mode: "contained-red"', () => {
    expect(
      render(<BaseButton mode="contained-red" label={'Base Button'} />),
    ).toMatchSnapshot()
  })
})
