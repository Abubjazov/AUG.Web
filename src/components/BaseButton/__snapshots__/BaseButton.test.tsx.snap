// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BaseButton > should render BaseButton default 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="_root_954613 _outlined_954613 "
        data-testid="base-button"
        type="button"
      >
        Base Button
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="_root_954613 _outlined_954613 "
      data-testid="base-button"
      type="button"
    >
      Base Button
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`BaseButton > should render BaseButton disabled 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="_root_954613 _outlined_954613 _disabled_954613"
        data-testid="base-button"
        disabled=""
        type="button"
      >
        Base Button
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="_root_954613 _outlined_954613 _disabled_954613"
      data-testid="base-button"
      disabled=""
      type="button"
    >
      Base Button
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`BaseButton > should render BaseButton disabled 2`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="_root_954613 _outlined_954613 _disabled_954613"
        data-testid="base-button"
        disabled=""
        type="button"
      >
        Base Button
      </button>
    </div>
    <div>
      <button
        class="_root_954613 _outlined_954613 _disabled_954613"
        data-testid="base-button"
        disabled=""
        type="button"
      >
        Base Button
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="_root_954613 _outlined_954613 _disabled_954613"
      data-testid="base-button"
      disabled=""
      type="button"
    >
      Base Button
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`BaseButton > should render BaseButton mode: "contained-blue" 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="_root_954613 _contained-blue_954613 "
        data-testid="base-button"
        type="button"
      >
        Base Button
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="_root_954613 _contained-blue_954613 "
      data-testid="base-button"
      type="button"
    >
      Base Button
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`BaseButton > should render BaseButton mode: "contained-red" 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="_root_954613 _contained-red_954613 "
        data-testid="base-button"
        type="button"
      >
        Base Button
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="_root_954613 _contained-red_954613 "
      data-testid="base-button"
      type="button"
    >
      Base Button
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`BaseButton > should render BaseButton mode: "outlined" 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="_root_954613 _outlined_954613 "
        data-testid="base-button"
        type="button"
      >
        Base Button
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="_root_954613 _outlined_954613 "
      data-testid="base-button"
      type="button"
    >
      Base Button
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`BaseButton > should render BaseButton mode: "outlined" 2`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="_root_954613 _outlined_954613 "
        data-testid="base-button"
        type="button"
      >
        Base Button
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="_root_954613 _outlined_954613 "
      data-testid="base-button"
      type="button"
    >
      Base Button
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`BaseButton > should render BaseButton skeleton 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="_root_954613 _skeleton_954613"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="_root_954613 _skeleton_954613"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`BaseButton > should render BaseButton skeleton 2`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="_root_954613 _skeleton_954613"
      />
    </div>
    <div>
      <div
        class="_root_954613 _skeleton_954613"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="_root_954613 _skeleton_954613"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`BaseButton > should render BaseButton skeleton 3`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="_root_954613 _skeleton_954613"
      />
    </div>
    <div>
      <div
        class="_root_954613 _skeleton_954613"
      />
    </div>
    <div>
      <div
        class="_root_954613 _skeleton_954613"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="_root_954613 _skeleton_954613"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`BaseButton > should render BaseButton skeleton 4`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="_root_954613 _skeleton_954613"
      />
    </div>
    <div>
      <div
        class="_root_954613 _skeleton_954613"
      />
    </div>
    <div>
      <div
        class="_root_954613 _skeleton_954613"
      />
    </div>
    <div>
      <div
        class="_root_954613 _skeleton_954613"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="_root_954613 _skeleton_954613"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
